llvm/test/CodeGen/AArch64/speculation-hardening-sls-blr.mir

```diff
@@ -46,13 +40,5 @@ body:             |
 
 
 ...
----
-name:            __llvm_slsblr_thunk_x8
-tracksRegLiveness: true
-body:             |
-  bb.0.entry:
-    liveins: $x8
 
-    BR $x8
```

Comment:
I think it is important to check that the body of the thunk inserted is as expected.
What is the motivation to remove that check, and only check that a thunk with the expected name was inserted, without checking its body?

Reply:
The above lines are **input** to `llc`, so I removed them to not conflict with the thunks that are inserted by the pass. Unfortunately, MIR syntax is quite verbose, so I added a few basic CHECK lines testing MIR output and duplicated the RUN lines to generate and check assembly output as well.

Now there is no separation between function-rewriting and thunk-inserting passes, and I assumed it is safe to expect that initially there are no thunks in the module. I guess the original version of thunk insertion logic would assert on pre-existing thunks as well (due to unknown thunk names with `.1` suffix).

Reply:
ah, yes, I was confused and didn't spot these were input lines....
I looked a bit more now at all of the existing SLS-hardening tests, and the content of the thunk is already tested in https://github.com/llvm/llvm-project/blob/f15266e9d5d87a291a515064f8f0dd5f0c412ed3/llvm/test/CodeGen/AArch64/speculation-hardening-sls.ll#L195

So, there isn't really a need to test for it separately here.
Apologies for giving you more work here!

I think it's fine for the test to go in as is; and it's equally fine for the test to go in without the CHECK lines for the content of the thunk. I'll leave it to you to make that decision.